src/lib/loaders/index.ts

```diff
@@ -39,12 +41,12 @@ export interface LoaderFactory {
     globalParams: any,
     pathAPIOptions: { headers: false } & APIOptions
   ): DynamicPathLoader<Body, PathGeneratorParams>
-  <Body = any, Headers = any>(
+  <Body = any, Headers = ResponseHeaders>(
```

Comment:
Previously the default type for the headers object was `any`, so `headers["x-total-count"]` was also `any` and passing it to the `arrayLength` property wouldn’t fail, even though it requires a `number`.

So I changed the default of headers to be an object where the value is a `string`, which then started failing on all these places where we were passing a `string` instead of a `number`.